src/Tmds.Ssh/SftpClient.cs

```diff
@@ -0,0 +1,156 @@
+
+        private ValueTask SendRequestAsync(Packet packet, SftpOperation operation)
+        {
+            /*
+                0: SSH_MSG_CHANNEL_DATA  (1)
+                1: RemoteChannel         (4)
+                5: DATA length           (4)
+                9: SFTP length           (4)
+                13: SFTP packet type     (1) ] // Already filled in.
+                14: SFTP request id      (4)
+            */
+            uint requestId = unchecked((uint)Interlocked.Increment(ref _requestId));
+            Span<byte> header = packet.PayloadHeader;
+            header[0] = (byte)MessageId.SSH_MSG_CHANNEL_DATA;
+            BinaryPrimitives.WriteUInt32BigEndian(header.Slice(1), _context.RemoteChannel);
+            int dataLength = (int)packet.PayloadLength - 9;
+            BinaryPrimitives.WriteUInt32BigEndian(header.Slice(5), (uint)dataLength);
+            BinaryPrimitives.WriteUInt32BigEndian(header.Slice(9), (uint)(dataLength - 4));
+            BinaryPrimitives.WriteUInt32BigEndian(header.Slice(14), requestId);
+
+            _operations.TryAdd(requestId, operation);
+
+            return _context.SendChannelDataAsync(packet, default(CancellationToken)); // TODO: cancellation
+        }
+
+        internal SftpClient(ChannelContext context)
+        {
+            _context = context;
+            _receiveLoopTask = ReceiveLoopAsync();
+            _operations = new ConcurrentDictionary<uint, SftpOperation>();
+        }
+
+        // public async Task InitAsync(CancellationToken ct)
+        // {
+        //     await _context.SftpInitMessageAsync(3, ct).ConfigureAwait(false);
+        //     // TODO add server negotiation in case server would have min. version < 3, but was able to do a version 3 aswell
+        //     int serverVersion = await _context.ReceiveServerVersionAsync("Failed to negotiate SFTP", ct).ConfigureAwait(false);
+
+        //     if (serverVersion != SftpVersion)
+        //         ThrowHelper.ThrowNotSupportedException("Server SFTP version is not supported");
+        // }
+
+        public void Dispose()
+        {
+            _context?.Dispose();
+        }
+
+        private async Task ReceiveLoopAsync()
+        {
+            try
+            {
+                MessageId messageId;
+                do
+                {
+                    using var packet = await _context.ReceivePacketAsync(ct: default).ConfigureAwait(false);
+                    messageId = packet.MessageId!.Value;
+
+                    if (messageId == MessageId.SSH_MSG_CHANNEL_DATA)
+                    {
+                        await HandleChannelData(packet.Move()).ConfigureAwait(false);
+                    }
+                    else
+                    {
+                        // Nothing yet
+                    }
+
+                } while (messageId != MessageId.SSH_MSG_CHANNEL_CLOSE);
+
+            }
+            catch (Exception e)
+            {
+                throw e;
```

Comment:
Let's add a TODO for dealing this exception.
The handling should avoid new operations being started, and an exception to be thrown for all on-going requests.